### DISCUSSION TITLE
github public PGP / GPG key server?

### DISCUSSION THREAD

**Discussion by jtmoon79** [community/community#36925] [Code Security]
tl;dr does github.com provide a public PGP key server?

I generated a PGP key using `gpg`. Then I added it to my github profile (_Settings_ -> _SSH and GPG keys_) .
Then I uploaded my GPG key to `keyserver.pgp.com`. That server then attempts to verify the email address of the key. However, the key uses my github private email address `123456+user@users.noreply.github.com`, which is not a receiving email address. So the keyserver `keyserver.pgp.com` cannot verify the user.
But github.com could internally route to that "email" address, and verify it is me. Then I could use github.com as a "receiving" key server, which allows more convenient key sharing.

Does github.com provide a PGP/GPG keyserver ?

**Comment by airtower-luna** [comment]
I'm not aware of a GitHub keyserver, but you can retrieve GPG keys for any user over the API: https://docs.github.com/en/rest/users/gpg-keys#list-gpg-keys-for-a-user

**Reply by spenserblack** [reply]
Also `github.com/USER.gpg` works if you just want to fetch a user's public key(s).
For example: https://github.com/jtmoon79.gpg

**Reply by joakime** [reply]
Is there a public key for the noreply@github.com or bots?  
They are using RSA key 4AEE18F83AFDEB23

**Reply by spenserblack** [reply]
@joakime I think that's the web-flow key: https://github.com/web-flow.gpg
You should also see most commits made over the web (like squash merges) are signed with that key.

**Reply by joakime** [reply]
@spenserblack cool, that did the trick, thanks!

**Comment by ossanna16** [comment]
Hi there @jtmoon79 and welcome to our community 😄. Thanks so much for asking a great question!
